src/js/containers/search/filters/awardID/AwardIDListContainer.jsx

```diff
@@ -0,0 +1,209 @@
+            this.parseAutocompleteAwardIDs(nextProps);
+        }
+    }
+
+    parseAutocompleteAwardIDs(awardIDProps) {
+        const values = [];
+        const fieldMap = {
+            piid: 'PIID',
+            fain: 'FAIN',
+            uri: 'URI',
+            parent_award__piid: 'PIID'
+        };
+
+        Object.keys(fieldMap).forEach((key) => {
+            const awardIDs = awardIDProps[key];
+            if (awardIDs && awardIDs.length > 0) {
+                awardIDs.forEach((item) => {
+                    if (item) {
+                        values.push({
+                            title: AwardIDFormatter.formatAwardID(item, fieldMap[key]),
+                            subtitle: fieldMap[key],
+                            data: Object.assign({}, item, {
+                                awardIDType: fieldMap[key]
+                            })
+                        });
+                    }
+                });
+            }
+        });
+
+        this.setState({
+            autocompleteAwardIDs: values
+        });
+    }
+
+    queryAutocompleteAwardIDs(input) {
+        this.setState({
+            noResults: false
+        });
+
+        // Only search if input is 2 or more characters
+        if (input.length >= 2) {
+            this.setState({
+                awardIDSearchString: input
+            });
+
+            if (this.awardIDSearchRequest) {
+                // A request is currently in-flight, cancel it
+                this.awardIDSearchRequest.cancel();
+            }
+
+            const awardIDSearchParams = {
+                fields: this.fields,
+                value: this.state.awardIDSearchString,
+                matched_objects: true
+            };
+
+            this.awardIDSearchRequest = SearchHelper.fetchAwardIDs(awardIDSearchParams);
+
+            this.awardIDSearchRequest.promise
+                .then((res) => {
+                    this.setState({
+                        noResults: !res.data.matched_objects.length
+                    });
+
+                    const data = res.data.matched_objects;
+                    let autocompleteData = {};
+
+                    // Remove custom awardIDType field from selectedAwardIDs
+                    // to enable object comparison with result data
+                    const selectedAwardIDs = [];
+                    this.props.selectedAwardIDs.toArray().forEach((item) => {
+                        selectedAwardIDs.push(_.omit(item, 'awardIDType'));
+                    });
+
+                    // Filter out any selectedAwardIDs that may be in the result set
+                    if (selectedAwardIDs && selectedAwardIDs.length > 0) {
+                        // We need to compare selectedAwardIDs to each result object,
+                        // as we're receiving a result object with 4 keys:
+                        // PIID, FAIN, URI, and PARENT_AWARD__PIID
+                        Object.keys(data).forEach((key) => {
```

Comment:
Why not just check the `id` value instead of the whole object? Isn't it supposed to be the DB primary key and thus unique?